slither/tools/read_storage/read_storage.py

```diff
@@ -0,0 +1,521 @@
+                logger.info(
+                    f"Contract {contract} not found in storage layout. It is possibly a parent contract"
+                )
+                continue
+
+            # Traverse the data structure of the target variable, return early if user does not provide key/ deep key/ struct var
+            if is_array(target_variable.type):
+                if key:
+                    info, type_to, slot, size, offset = _find_array_slot(
+                        target_variable, slot, key, deep_key=deep_key, struct_var=struct_var
+                    )
+                    log += info
+                else:
+                    value = get_storage_data(web3, checksum_address, slot).hex()
+                    int_slot = int.from_bytes(slot, byteorder="big")
+                    return (
+                        int_slot,
+                        value,
+                        type_to,
+                    )
+
+            elif is_user_defined_type(target_variable.type):
+                if struct_var:
+                    var_log_name = struct_var
+                    elems = target_variable.type.type.elems_ordered
+                    info, type_to, slot, size, offset = _find_struct_var_slot(
+                        elems, slot, struct_var
+                    )
+                    log += info
+                else:
+                    value = get_storage_data(web3, checksum_address, slot).hex()
+                    int_slot = int.from_bytes(slot, byteorder="big")
+                    return (
+                        int_slot,
+                        value,
+                        type_to,
+                    )
+
+            elif is_mapping(target_variable.type):
+                if key:
+                    info, type_to, slot, size, offset = _find_mapping_slot(
+                        target_variable, slot, key, struct_var=struct_var, deep_key=deep_key
+                    )
+                    log += info
+                else:
+                    value = get_storage_data(web3, checksum_address, slot).hex()
+                    int_slot = int.from_bytes(slot, byteorder="big")
+                    return (
+                        int_slot,
+                        value,
+                        type_to,
+                    )
+
+            else:  # elementary type
+                type_to = target_variable.type.name
+
+            hex_bytes = get_storage_data(web3, checksum_address, slot)
+            int_slot = int.from_bytes(slot, byteorder="big")
+
+            # Account for storage packing
+            offset_hex_bytes = get_offset_value(hex_bytes, offset, size)
+            value = coerce_type(type_to, offset_hex_bytes)
+
+            log += f"\nName: {var_log_name}\nType: {type_to}\nValue: {value}\nSlot: {int.from_bytes(slot, byteorder='big')}\n"
+            logger.info(log)
+
+            return int_slot, value, type_to
+
+    if not found:
```

Comment:
It looks like we can remove `found` and raise a exception if we reached this part of the code, is that correct?

Reply:
that would also avoid having an undefined return